docs/requirements/requirements.md

```diff
@@ -0,0 +1,103 @@
+# Hardware requirements of TwPM module
+
+## Acronyms
+
+* Legend:
+* PTP – Platform TPM Profile
+* CRB – Command Response Buffer interface
+* DDWG – Device Driver’s Writers Guide
+* Certification PP – Certification Protection Profile
+* TIS – TPM Interface Specification
+
+## Requirements
+
+### Power supply
+
+#### Power supply voltage
+
+1. TPM MUST support a supply voltage of 1.8V or 3.3V.
+2. TPM MAY support supply and I/O voltages of both 1.8V and 3.3V.
+3. TPM MAY support other supply voltages.
+
+Source: [tcg_client_v13r27] l.1373-1375
+
+#### Power supply max current
+
+Maximum power supply taken by TPM MUST be 250mA.
+
+### SPI Interface
+
+#### Frequency
+
+Default clock frequency for PC Client platforms is defined to be 24MHz.
+Clock frequency signal, minimum, is 10MHz.
+
+Source: [tcg_client_v13r27] Chapter 6.4.1, line 1331
+
+#### Voltage levels
+
+1. TPM MUST support a supply and I/O voltage of 1.8V or 3.3V
+2. TPM MAY support supply and I/O voltages of both 1.8 and 3.3V.
+3. TPM MAY support other supply and I/O voltages. 
+
+Source: [tcg_client_v13r27] l.1373-1375
+
+#### Power supply
+
+
+#### Power sequence
+
+  * TPM_Init (LRESET#/SPI_RST#) signal connected to platform CPU Reset signa
+  * TPM main power pins (3V) must be connected such that the TPM is powered
+  during ACPI states S0-S2 and may be powered in S3-S5
+  * Optional implementation VBAT and/or 3VSB pins
+
+Source: [tcg_client_v13r27], Chapter 6.7.2
+
+#### Command Duration
+
+Command duration MUST be less than 750ms.
+
+Source: [tck_client_v13r27], chapter 5.6.5
+
+#### Timeout
+
+Four kind of timeouts:
+ * TIMEOUT_A - default value 750ms
+ * TIMEOUT_B - default value 2s
+ * TIMEOUT_C - default value 750ms
+ * TIMEOUT_D - default value 750ms
+
+Source: [tck_client_v13r27], chapter 5.6.6
+
+#### Reset timing
+ 
+* Within 500 microseconds of the completion of TPM_Init:
+ all fields within all TPM_ACCESS_x registers must be valid logical level
+ as indicated by the tpmRegValidSts field;
+
+* Within 30 milliseconds of the completion of TPM_Init:
+ all fields within the access register and all other registers MUST return
+with the state of their fields valid.
+
+* The TPM MUST be ready to receive a command in 30ms
+
+Source: [tcg_client_v13r27], chapter 6.6 
+
+
+[TCG PC Client, v1.3, rev 27][tcg_client_v13r27]: https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClientTPMInterfaceSpecification_TIS__1-3_27_03212013.pdf
+[TCG TPM I2C Interface Specification][tcg_i2cspec_v200rev1]https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-I2C-Interface-Specification-v1.00.pdf
+
+
+# TCG PC Client Platform Physical Presence Interface Specification
```

Comment:
1. I don't think this should be H1 (`#`) level.
2. Some rendering engines don't like headers immediately followed by other text, please add empty lines before and after headers, here and few lines down for `Bibliography`.

Reply:
Ad.2

Checks as this are automated by pre-commit, so threre is no need to waste reviewer's time with such repeateable comments:

```
docs/requirements/requirements.md:92 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "# TCG PC Client Platform Physical Presence Interface Specification"]
```

Reply:
@krystian-hebel sure, corrected.

Reply:
@macpijan I've configured and run. Currently, I have the result, during precommit and commit:
```
$ pre-commit run --all-files
check for added large files..............................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
detect private key.......................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
mixed line ending........................................................Passed
yamllint.................................................................Passed
markdownlint.............................................................Passed
markdownlint-fix.........................................................Passed
codespell................................................................Passed
```

```
$ git commit
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
detect private key.......................................................Passed
mixed line ending........................................................Passed
yamllint.............................................(no files to check)Skipped
markdownlint.........................................(no files to check)Skipped
markdownlint-fix.....................................(no files to check)Skipped
Conform..................................................................Failed
- hook id: conform
- exit code: 1

POLICY        CHECK                        STATUS        MESSAGE                                   
commit        Header Length                PASS          Header is 49 characters                   
commit        Imperative Mood              PASS          Commit begins with imperative verb        
commit        Header Last Character        PASS          Header last character is valid            
commit        DCO                          FAILED        Commit does not have a DCO                
commit        GPG                          PASS          GPG signature found                       
commit        Spellcheck                   PASS          Commit contains 0 misspellings            
Error: 1 or more policy failed

codespell................................................................Passed
```

Reply:
@3mdeb-karolzmyslowski please read what you get her, DCO is invalid: https://github.com/apps/dco

AFAIK we have recommended configuration for gitconfig, which seem to not be used here

https://github.com/pietrushnic/dotfiles/blob/master/gitconfig#L2

I always use `git ci`

Reply:
@pietrushnic @macpijan solved, thanks for notes.